testing/test_assertrewrite.py

```diff
@@ -180,6 +180,27 @@ class X(object):
 
         assert getmsg(f, {"cls": X}) == "assert cls == 42"
 
+    def test_dont_rewrite_if_hasattr_fails(self):
```

Comment:
Interesting, this test passes on `features` just fine on `py36`, but indeed fails on `py27`. 👍

Reply:
This is due to the hasattr changes in python 3 - - perhaps we should make a note here so this gets cleaned up when dropping python 2

Reply:
Can you explain a bit more about what we want to make a note of? As I understand it, this fixes behaviour that we'd only see after [python 3.2](https://docs.python.org/dev/whatsnew/3.2.html), when `hasattr` stopped catching errors that aren't `AttributeError`.

Reply:
oh right, I was thinking the code was in the other direction -- carry on this is fine 👍